src/schema/v2/lot.ts

```diff
@@ -0,0 +1,179 @@
+          floorSellingPriceCents
+          onlineSellingToBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          floorWinningBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          soldStatus
+        }
+      }
+    `,
+    variables: {
+      ids: saleArtworkIds,
+    },
+  }).then((res: FetchResponse) => res.json())
+
+  const { data: causalityData, errors: causalityErrors } = lotStatesResponse
+
+  // If the causality request failed for some reason, throw its errors.
+  if (causalityErrors) {
+    const errors = causalityErrors.reduce((acc, ce) => {
+      return acc + " " + ce["message"]
+    }, "From causality: ")
+    throw new GraphQLError(errors)
+  } else {
+    // make a map of lot state data
+    const lotDataMap = causalityData.lots.reduce((acc, lot) => {
+      acc[lot.internalID] = lot
+      return acc
+    }, {})
+
+    // place lotData in context for retrieval at stitching stage
+    context.lotDataMap = lotDataMap
+
+    // create nodes from sale artworks
+    const availableLotDataIds = Object.keys(lotDataMap)
+    const nodes = watchedSaleArtworks.reduce((acc, saleArtwork) => {
+      if (!availableLotDataIds.find((id) => id === saleArtwork._id)) {
+        console.warn(`lot state for ${saleArtwork._id} not found - skipping`)
+        return acc
+      } else {
+        return [...acc, { _id: saleArtwork._id, saleArtwork }]
+      }
+    }, [])
+
+    return {
+      totalCount,
+      ...connectionFromArray(nodes, connectionOptions),
+    }
+  }
+}
+
+/**
+ * An extension schema to add the lot type at the causality stitching stage.
+ */
+export const stitchingLotExtensionSchema = gql`
+  # A unified auction lot with data from our auctions bidding engine.
+  extend type Lot {
+    # The current auction state of the lot.
+    lot: AuctionsLotState!
+  }
+`
+
+/**
+ * A Resolver for use with causality's stitching file, but kept here because
+ * it is tied to assumptions about our watchedLotConnection resolver's
+ * use of context
+ */
+export const stitchedCausalityLotResolver = {
+  Lot: {
+    lot: {
+      resolve: (root, _args, context, _info) => {
+        const { internalID } = root
+
+        // resolve lot if available via context (eg watchedLotConnection resolver)
+        const lotState = context.lotDataMap?.[internalID]
+        if (lotState) {
+          return lotState
+        }
+
+        throw new GraphQLError(`Lot state for ${internalID} missing`)
+      },
+    },
+  },
+}
+
+export const auctionLotConnection = connectionWithCursorInfo({
+  nodeType: LotType,
+})
+
+export const watchedLotConnection = {
+  description: "A list of lots a user is watching",
+  type: auctionLotConnection.connectionType,
+  args: pageable(),
+  resolve: resolveWatchedLotConnection,
+}
```

Comment:
And here is a `watchedLotConnection` ready to be exported to our local schema (as `Me.watchedLotConnection`)